src/hotspot/share/runtime/os.cpp

```diff
@@ -1528,6 +1528,56 @@ bool os::set_boot_path(char fileSep, char pathSep) {
   return false;
 }
 
+static char* _image_release_file_content = nullptr;
+
+void os::read_image_release_file() {
+  const char* home = Arguments::get_java_home();
+  int rfile_len = (int)strlen(home) + 10;
+
+  char* release_file = (char*) os::malloc(rfile_len, mtInternal);
+  if (release_file) {
+    os::snprintf(release_file, rfile_len, "%s/release", home);
+
+    if (_image_release_file_content == nullptr) {
+      FILE *file = fopen(release_file, "rb");
+      if (!file) {
+        free(release_file);
+        return;
+      }
+      fseek(file, 0, SEEK_END);
+      long sz = ftell(file);
+      if (sz == -1) {
+        free(release_file);
+        return;
+      }
+      fseek(file, 0, SEEK_SET);
+
+      _image_release_file_content = (char*) os::malloc(sz + 1, mtInternal);
+      if (!_image_release_file_content) {
+        fclose(file);
+        free(release_file);
+        return;
+      }
+
+      size_t elements_read = fread(_image_release_file_content, 1, sz, file);
+      if (elements_read < (size_t)sz) _image_release_file_content[elements_read] = '\0';
+      _image_release_file_content[sz] = '\0';
```

Comment:
Shouldn't this be in an else?

Reply:
Yeah we probably (most likely) only need one '\0' .